src/main/java/org/example/MessageServiceImpl.java

```diff
@@ -3,40 +3,35 @@
 import java.time.LocalDate;
 import java.time.LocalDateTime;
 import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
 import java.time.temporal.ChronoUnit;
 
-public class MessageServiceImpl implements MessageService{
+public class MessageServiceImpl implements MessageService {
 
-  private static final LocalDate baselineDate = LocalDate.parse("2012-08-06",
-      DateTimeFormatter.ofPattern("yyyy-MM-dd"));
+  // format for ISO 8601
+  private static final DateTimeFormatter isoFormat = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+  private static final LocalDate curiosityLandingDate = LocalDate.parse("2012-08-06",
+      isoFormat);
 
   @Override
   public String getHello() {
     return "Hello World Jersey and HK2 ";
   }
 
   @Override
-  public int curiosityDateDateConversion(String processingDate) {
-    float diff_In_Days ;
-    LocalDate earthDate;
-    // format for ISO 8601
-    DateTimeFormatter format = DateTimeFormatter.ofPattern("yyyy-MM-dd");
+  public int  curiosityDateConversion(String date) {
 
-    if(processingDate == null){
-      // assign current date
-      processingDate = format.format(LocalDateTime.now());
+    try {
+      LocalDate earthDate = LocalDate.parse(date, isoFormat);
+      float diff_In_Days = ChronoUnit.DAYS.between(curiosityLandingDate, earthDate);
+      return (int) Math.round(diff_In_Days * 86400 / 88775.245) ;
```

Comment:
```suggestion
      return Math.round(diff_In_Days * 86400 / 88775.245) ;
```
I guess casting to int is excessive here. `Math.round` from `float` should already return an `int`.